app/src/main/java/com/juliazluo/www/pokedex/PokemonHandler.java

```diff
@@ -0,0 +1,82 @@
+package com.juliazluo.www.pokedex;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+/**
+ * Created by julia on 2017-02-14.
+ */
+
+public class PokemonHandler {
+
+    private Pokedex pokedex;
+    private ArrayList<Pokedex.Pokemon> pokemonList;
+    private ArrayList<String> pokemonTypes;
+
+    public PokemonHandler() {
+        pokedex = new Pokedex();
+        pokemonList = pokedex.getPokemon();
+        pokemonTypes = populateTypes();
+    }
+
+    public ArrayList<String> populateTypes() {
+        ArrayList<String> types = new ArrayList<>();
+        for (Pokedex.Pokemon pokemon : pokemonList) {
+            for (int i = 0; i < pokemon.types.length; i++) {
+                if (!types.contains(pokemon.types[i])) {
+                    types.add(pokemon.types[i]);
+                }

```

Comment:
you're going to do this everytime you make a handler? better to just make an array of types, there are only like 15 or something